lib/dropbox_api/endpoints/content_upload.rb

```diff
@@ -10,10 +10,16 @@ def build_request(params, content)
       # TODO: It would be better to have a stream object on which we can call
       #       #read, rather than the full file content.
       body = content
+      length = if body.respond_to?(:length)
+        body.length
+      elsif body.respond_to?(:stat)
+        body.stat.size
+      end
       headers = {
         'Dropbox-API-Arg' => JSON.dump(params),
-        'Content-Type' => 'application/octet-stream'
+        'Content-Type' => 'application/octet-stream',
       }
+      headers['Content-Length'] = length.to_s if length
```

Comment:
This may be a bit nitpicking, but wouldn't this be much simpler if written this way:
```
      if body.respond_to?(:length)
        headers['Content-Length'] = length
      elsif body.respond_to?(:stat)
        headers['Content-Length'] = body.stat.size
      end
```
Saves you from having a lengh variable, also checking if it's nil...